libbeat/docs/config-file-format.asciidoc

```diff
@@ -0,0 +1,355 @@
+[[config-file-format]]
+== Config file format
+
+Beats config files are based on http://www.yaml.org[YAML], a file format that is
+easier to read and write then other common data formats like XML or JSON.
+
+In beats all YAML files start with a "dictionary" (often called a "hash"). In
+addition to dictionaries, YAML also supports lists, numbers, strings, and many
+other data types. All members of lists and dictionaries must have the same
+indentation level.
+
+Dictionaries are represented by simple `key: value` pairs all having the same
+indentation level. The colon after `key` must be followed by a space.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+name: John Doe
+age: 34
+country: Canada
+------------------------------------------------------------------------------
+
+Lists are introduced by dashes `- `. All list members will be lines beginning
+with `- ` at the same indentation level.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+- Red
+- Green
+- Blue
+------------------------------------------------------------------------------
+
+Lists and dictionaries are used in beats to build structured configurations.
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+filebeat:
+  prospectors:
+    - input_type: log
+      paths:
+        - /var/log/*.log
+      multiline:
+        pattern: '^['
+        match: after
+------------------------------------------------------------------------------
+
+Lists and dictionaries can also be represented in abbreviated form. Abbreviated
+form is somewhat similar to JSON using `{}` for dictionaries and `[]` for lists:
+
+["source","yaml",subs="attributes"]
+------------------------------------------------------------------------------
+person: {name: "John Doe", age: 34, country: "Canada"}
+colors: ["Red", "Green", "Blue"]
+------------------------------------------------------------------------------
+
+
+[[config-file-format-namespacing]]
+=== Namespacing
+
+All settings bases on dictionaries and lists are collapsed into "namespaced"
```

Comment:
Does it really matter who/where this happens from technical point of view?

TBH they are not really collapsed, but split up in structure (technically the exact opposite) + configs are always accessible using full path or sub-pathes.

That is, the 'collapsing' part is more of an idea for users.